src/ert/_c_wrappers/job_queue/forward_model.py

```diff
@@ -63,21 +65,106 @@ def formatted_fprintf(
         run_id,
         path,
         data_root,
-        global_args: SubstitutionList,
+        iens: int,
+        itr: int,
+        substituter: "Substituter",
         env_varlist: EnvironmentVarlist,
+        filename: str = "jobs.json",
     ):
-        self._formatted_fprintf(run_id, path, data_root, global_args, env_varlist)
+        def substitute(job, string):
+            if string is not None:
+                return substituter.substitute(
+                    job.get_private_args().substitute(string), iens, itr
+                )
+            else:
+                return string
+
+        def nulled_int(integer: int) -> Optional[int]:
```

Comment:
may I suggest a somwhat longer and less elegant, but more descriptive name? `positive_or_null_int`? don't feel very strongly about it though, so feel free to disregard the suggestion.

Reply:
Fixed